server/svix-server/src/v1/endpoints/attempt.rs

```diff
@@ -65,6 +65,95 @@ impl From<messageattempt::Model> for MessageAttemptOut {
     }
 }
 
+// FIXME: Contains all members of a [`v1::endpoints::message::MessageOut`], so find a way to
+// #[serde(flatten)]` a [`super::message::MessageOut`] in cleanly. An attempt was made, but it would
+// require a custom [`ModelOut`] impl and another impl of [`sea_orm::TryGettable`] on the
+// [`super::message::MessageOut`] type in order for [`FromQueryResult`] to work.
+
+/// A model containing information on a given message plus additional fields on the last attempt for
+/// that message.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, ModelOut, FromQueryResult)]
+#[serde(rename_all = "camelCase")]
+struct AttemptedMessageOut {
+    event_type: EventTypeName,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_id: Option<MessageUid>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    channels: Option<EventChannelSet>,
+    payload: serde_json::Value,
+    id: MessageId,
+    #[serde(rename = "timestamp")]
+    created_at: DateTimeWithTimeZone,
+    status: MessageStatus,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    next_attempt: Option<DateTimeWithTimeZone>,
+}
+
+/// Additional parameters (besides pagination) in the query string for the "List Attempted Messages"
+/// enpoint.
+#[derive(Debug, Deserialize, Validate)]
+pub struct ListAttemptedMessagesQueryParameters {
+    #[validate]
+    channel: Option<EventChannel>,
+    status: Option<MessageStatus>,
+}
+
+/// Fetches a list of [`AttemptedMessageOut`]s associated with a given app and endpoint.
+async fn list_attempted_messages(
+    Extension(ref db): Extension<DatabaseConnection>,
+    ValidatedQuery(mut pagination): ValidatedQuery<Pagination<MessageId>>,
+    ValidatedQuery(ListAttemptedMessagesQueryParameters { channel, status }): ValidatedQuery<
+        ListAttemptedMessagesQueryParameters,
+    >,
+    Path((_app_id, endp_id)): Path<(ApplicationId, EndpointId)>,
+    AuthenticatedApplication {
+        permissions: _,
+        app: _,
+    }: AuthenticatedApplication,
+) -> Result<Json<ListResponse<AttemptedMessageOut>>> {
+    let limit = pagination.limit;
+    let iterator = pagination.iterator.take();
+
+    let mut msg_and_dest = messagedestination::Entity::secure_find_by_endpoint(endp_id)
+        .order_by_desc(messagedestination::Column::MsgId)
+        .limit(limit + 1)
+        .column_as(message::Column::Id, "id")
+        .column_as(message::Column::EventType, "event_type")
+        .column_as(message::Column::Uid, "event_id")
+        .column_as(message::Column::Channels, "channels")
+        .column_as(message::Column::Payload, "payload")
+        .column_as(message::Column::CreatedAt, "created_at")
```

Comment:
This is painfully manual, why does it have to be like this? What happened to how it was before?

Reply:
So, with the previous version it didn't actually select  any of the `message` columns and panicked unable to fill the struct. This is the only way I found with a quick glance at the SeaORM documentation, but I'll look for a better solution before we merge.